crates/biome_js_analyze/src/lint/nursery/use_new_for_builtins.rs

```diff
@@ -0,0 +1,273 @@
+        };
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Use "<Emphasis>{use_this}{name}"()"</Emphasis>" instead of "<Emphasis>{instead_of}{name}"()"</Emphasis>"."
+            },
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+
+        match node {
+            JsNewOrCallExpression::JsNewExpression(node) => action_remove_new(ctx, node),
+            JsNewOrCallExpression::JsCallExpression(_) => None,
+        }
+    }
+}
+
+fn extract_callee_and_rule(
+    node: &JsNewOrCallExpression,
+) -> Option<(AnyJsExpression, BuiltinCreationRule)> {
+    match node {
+        JsNewOrCallExpression::JsNewExpression(node) => {
+            let callee = node.callee().ok()?;
+
+            Some((callee, BuiltinCreationRule::MustNotUseNew))
+        }
+        JsNewOrCallExpression::JsCallExpression(node) => {
+            let callee: AnyJsExpression = node.callee().ok()?;
+
+            Some((callee, BuiltinCreationRule::MustUseNew))
+        }
+    }
+}
+
+fn convert_new_expression_to_call_expression(expr: &JsNewExpression) -> Option<JsCallExpression> {
+    let new_token = expr.new_token().ok()?;
+    let mut callee = expr.callee().ok()?;
+    if new_token.has_leading_comments() || new_token.has_trailing_comments() {
+        callee = callee.prepend_trivia_pieces(chain_trivia_pieces(
+            new_token.leading_trivia().pieces(),
+            new_token.trailing_trivia().pieces(),
+        ))?;
+    }
+    Some(make::js_call_expression(callee, expr.arguments()?).build())
+}
+
+fn action_remove_new(
+    ctx: &RuleContext<UseNewForBuiltins>,
+    node: &JsNewExpression,
+) -> Option<JsRuleAction> {
+    let call_expression = convert_new_expression_to_call_expression(node)?;
+    let mut mutation = ctx.root().begin();
+    mutation.replace_node::<AnyJsExpression>(node.clone().into(), call_expression.into());
+    Some(JsRuleAction {
+        category: ActionCategory::QuickFix,
+        applicability: Applicability::MaybeIncorrect,
+        message: markup! { "Remove "<Emphasis>"new"</Emphasis>"." }.to_owned(),
+        mutation,
+    })
+}
+
+fn _convert_call_expression_to_new_expression(expr: &JsCallExpression) -> Option<JsNewExpression> {
+    let new_token = token_decorated_with_space(JsSyntaxKind::NEW_KW);
+
+    let mut callee = expr.callee().ok()?;
+    if new_token.has_leading_comments() || new_token.has_trailing_comments() {
+        callee = callee.prepend_trivia_pieces(chain_trivia_pieces(
+            new_token.leading_trivia().pieces(),
+            new_token.trailing_trivia().pieces(),
+        ))?;
+    }
+
+    // TODO. make::js_new_expression currently does not accept arguments.
+    // To fix that js.ungram needs to be updated, but that breaks a lot of rules.
+    Some(make::js_new_expression(new_token, callee).build())
```

Comment:
I tried changing ungram, but that broke bunch of things. Removing new code fix works, but adding is currently disabled, because I am not sure what else to use besides `make::js_new_expression` here.

Should I remove these unused functions entirely?

Reply:
I am not sure I understand the issue, would you mind providing more context/pseudo code?

We haven't had issues with the codegen, because these functions are exactly created from the grammar. It's possible you're trying to create a CST that isn't the same as the grammar, or maybe you aren't providing the correct nodes.

Reply:
I found out I can do `make::js_new_expression(new_token, callee).with_arguments(expr.arguments().ok()?)`

---------
Previous thinking:
My current fix generates:
```js
// input
Date(1)
// output
new Date
```
It looses arguments part.
`make::js_new_expression` does not accept arguments param
<img width="262" alt="image" src="https://github.com/biomejs/biome/assets/8914032/659361f3-0c30-4d27-9232-a89e631ea619">
because (as far as I understand) in ungram this field is marked with question mark.
<img width="312" alt="image" src="https://github.com/biomejs/biome/assets/8914032/a159c74b-e210-4a27-9e26-cf41e7dc1575">

Analogical `make::js_call_expression` does have it.
<img width="317" alt="image" src="https://github.com/biomejs/biome/assets/8914032/acf2153c-d089-4a4b-a660-49f8fac357f6">